packages/cdk/directives/src/initial-focus/initial-focus.directive.ts

```diff
@@ -16,15 +19,23 @@ export class InitialFocusDirective implements AfterViewInit, OnDestroy {
     }
 
     public ngAfterViewInit(): void {
-        this.ngZone.runOutsideAngular((): void => {
-            window.clearInterval(this.timeoutId!);
-            // eslint-disable-next-line no-restricted-properties
-            this.timeoutId = window.setInterval((): void => this.focus(), this.focusDelay);
-        });
+        this.decideAndTryToFocus();
     }
 
     public ngOnDestroy(): void {
-        window.clearInterval(this.timeoutId!);
+        this.unsubscribe$.next(true);
+    }
+
+    private decideAndTryToFocus(): void {
+        if (!this.disabled && !this.isFocused()) {
+            this.ngZone.runOutsideAngular((): void => {
```

Comment:
I'll be waiting before you add outsideZone operator

Reply:
I'd better have it fixed at a refactoring task
initial-focus runs smoothly 
need it for another task

Reply:
Ok https://github.com/Angular-RU/angular-ru-sdk/issues/1148